manubot/cite/csl_item.py

```diff
@@ -145,64 +145,121 @@ def clean(self, prune: bool = True):
             self.validate_against_schema()
         return self
 
+    @property
+    def note(self) -> str:
+        """
+        Return the value of the "note" field as a string.
+        If "note" key is not set, return empty string.
+        """
+        return str(self.get('note') or '')
 
-def assert_csl_item_type(x):
-    if not isinstance(x, CSL_Item):
-        raise TypeError(
-            f'Expected CSL_Item object, got {type(x)}')
+    @note.setter
+    def note(self, text: str):
+        if text:
+            self['note'] = text
+        else:
+            self.pop('note', None)
```

Comment:
@epogrebnyak I haven't used a property setter before, but I think this is right.

Reply:
Niether did I, just learnt recently. What is the behavior idea behind else?

Reply:
Added a comment in https://github.com/manubot/manubot/pull/157/commits/97df68a905e531ca0dcdfe85c9d67a3853eb2516 to make this more clear:

> if text is None or an empty string, remove the "note" field